router/core/graphql_handler.go

```diff
@@ -375,9 +375,17 @@ func addErrorToSpan(ctx context.Context, err error) {
 	rtrace.AttachErrToSpan(span, err)
 }
 
-func propagateSubgraphErrors(ctx *resolve.Context) {
+func propagateSubgraphErrors(ctx *resolve.Context, requestLogger *zap.Logger) {
+	// Get subgraph errors
 	err := ctx.SubgraphErrors()
+	// Check if there are subgraph errors
+	if err != nil {
```

Comment:
That's comedy :smile: In case of an error, we early return, so we will never process the error.

Reply:
Here is no defer inside. When `err` is not nil the function will return immediately but we want to pass it to `addErrorToSpan` and log it.

Reply:
I've edited as you requested. I agree that there isn't any defer inside this function but the call of this one is defered and the `addErrorToSpan` is stopping early because error is nil. I was just talking about this :) .